diff -p -U8 xchat-2.8.8/src/fe-gtk/plugin-tray.c.libnotifyso4 xchat-2.8.8/src/fe-gtk/plugin-tray.c
--- xchat-2.8.8/src/fe-gtk/plugin-tray.c.libnotifyso4	2011-04-07 15:08:10.620286679 -0700
+++ xchat-2.8.8/src/fe-gtk/plugin-tray.c	2011-04-07 15:08:40.830908993 -0700
@@ -144,17 +144,17 @@ libnotify_cleanup (void)
 
 static gboolean
 libnotify_notify_new (const char *title, const char *text, GtkStatusIcon *icon)
 {
 	void *noti;
 
 	if (!nn_mod)
 	{
-		nn_mod = g_module_open ("libnotify", G_MODULE_BIND_LAZY);
+		nn_mod = g_module_open ("libnotify.so.4", G_MODULE_BIND_LAZY);
 		if (!nn_mod)
 		{
 			nn_mod = g_module_open ("libnotify.so.1", G_MODULE_BIND_LAZY);
 			if (!nn_mod)
 				return FALSE;
 		}
 
 		if (!g_module_symbol (nn_mod, "notify_init", (gpointer)&nn_init))
