--- xchat-2.6.6/src/common/xchat.c.orig	2006-08-02 23:07:47.000000000 -0400
+++ xchat-2.6.6/src/common/xchat.c	2006-08-02 23:10:16.000000000 -0400
@@ -578,7 +578,7 @@
 #ifdef WIN32
 static char defaultconf_urlhandlers[] =
 	"NAME Connect as IRC server\n"		"CMD url %s\n\n";
-#else
+#elif 0
 static char defaultconf_urlhandlers[] =
 	"NAME SUB\n"								"CMD Epiphany...\n\n"\
 		"NAME Open\n"							"CMD !epiphany '%s'\n\n"\
@@ -626,6 +626,9 @@
 		"NAME Ping\n"							"CMD !"XTERM"ping -c 4 %s\n\n"\
 	"NAME ENDSUB\n"							"CMD \n\n"\
 	"NAME Connect as IRC server\n"		"CMD url %s\n\n";
+#else
+static char defaultconf_urlhandlers[] =
+	"NAME Open URL in Web Browser\n"	"CMD !gnome-open '%s'\n\n";
 #endif
 #endif
 
