diff -Naru xchat-2.4.3.orig/configure xchat-2.4.3/configure
--- xchat-2.4.3.orig/configure	2005-03-29 08:47:31.000000000 +0300
+++ xchat-2.4.3/configure	2005-04-14 23:20:09.000000000 +0300
@@ -30054,11 +30054,11 @@
 _ACEOF
 
 	cat >>confdefs.h <<_ACEOF
-#define XCHATLIBDIR "${prefix}/lib/xchat"
+#define XCHATLIBDIR "${libdir}/xchat"
 _ACEOF
 
 	cat >>confdefs.h <<_ACEOF
-#define XCHATSHAREDIR "${prefix}/share/xchat"
+#define XCHATSHAREDIR "${datadir}/xchat"
 _ACEOF
 
 fi
diff -Naru xchat-2.4.3.orig/configure.in xchat-2.4.3/configure.in
--- xchat-2.4.3.orig/configure.in	2005-03-29 08:45:59.000000000 +0300
+++ xchat-2.4.3/configure.in	2005-04-14 23:20:47.000000000 +0300
@@ -641,8 +641,8 @@
 	AC_DEFINE_UNQUOTED(XCHATSHAREDIR, "${ac_default_prefix}/share/xchat")
 else
 	AC_DEFINE_UNQUOTED(PREFIX, "${prefix}")
-	AC_DEFINE_UNQUOTED(XCHATLIBDIR, "${prefix}/lib/xchat")
-	AC_DEFINE_UNQUOTED(XCHATSHAREDIR, "${prefix}/share/xchat")
+	AC_DEFINE_UNQUOTED(XCHATLIBDIR, "${libdir}/xchat")
+	AC_DEFINE_UNQUOTED(XCHATSHAREDIR, "${datadir}/xchat")
 fi
 
 dnl for plugins/xxx/Makefile.am
