diff -ur xchat-2.8.2/src/common/cfgfiles.c xchat-2.8.2-disable-tray-icon-by-default/src/common/cfgfiles.c
--- xchat-2.8.2/src/common/cfgfiles.c	2007-03-26 04:46:05.000000000 +0200
+++ xchat-2.8.2-disable-tray-icon-by-default/src/common/cfgfiles.c	2007-06-02 20:48:21.000000000 +0200
@@ -663,7 +663,7 @@
 	prefs.autoopendccchatwindow = 1;
 	prefs.userhost = 1;
 	prefs.gui_url_mod = 4;	/* ctrl */
-	prefs.gui_tray = 1;
+	prefs.gui_tray = 0;
 	prefs._gui_tray_blink = 0xfc920; /* 1111 1100 1001 0010 0000 */
 	prefs.mainwindow_save = 1;
 	prefs.bantype = 2;
