--- xchat-2.4.3/src/fe-gtk/fkeys.c.orig	2005-03-10 20:32:28.000000000 -1000
+++ xchat-2.4.3/src/fe-gtk/fkeys.c	2005-05-15 21:03:27.000000000 -1000
@@ -288,6 +288,11 @@
 				kb->next = keys_root;
 				keys_root = kb;
 			}
+			/* make sure this key event is filtered in the Input Method. */
+			if (gtk_im_context_filter_keypress (GTK_ENTRY (wid)->im_context, evt)) {
+				GTK_ENTRY (wid)->need_im_reset = TRUE;
+				return TRUE;
+			}
 			/* Run the function */
 			n = key_actions[kb->action].handler (wid, evt, kb->data1,
 															 kb->data2, sess);
